### PR TITLE
chore: support deploying to Openshift and add gitlab pipeline

### PR DESCRIPTION
**Pull request by ciarams87**
### Proposed changes
 Add changes to support deploying to Openshift, including new Makefile targets and a RedHat specific Dockerfile. The `openshift-deploy` Make target requires yq to be installed (see https://github.com/mikefarah/yq). This PR also addresses CVE-2020-29652 and adds a gitlab pipeline for testing.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-ingress-operator/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [ ] I have rebased my branch onto master
- [ ] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -117,6 +121,11 @@ deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in
 	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
 	$(KUSTOMIZE) build config/default | kubectl apply -f -
 
+openshift-deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	VAL="${RH_RBAC_IMAGE}" yq e '.spec.template.spec.containers[0].image = strenv(VAL)' -i config/default/manager_auth_proxy_patch.yaml
```
Comment (lucacome) [inline Makefile:125]: Is `yq` supposed to be `jq`? If so are we assuming it's installed by default?

File: openshift.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+COPY main.go main.go
+COPY api/ api/
+COPY controllers/ controllers/
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-s -w -X main.version=${VERSION}" -a -o manager main.go
+
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
+WORKDIR /
+COPY --from=builder --chown=65532:65532 /workspace/manager .
```
Comment (lucacome) [inline openshift.Dockerfile:23]: I think we need to remove `chown`? `65532` is not in UBI.

**Review by lucacome — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (soneillf5) [reply Makefile:125]: I think yq is to yaml as jq is to json

**Review by soneillf5 — APPROVED** [review]

File: openshift.Dockerfile (hunk 2 above)
Comment (ciarams87) [reply openshift.Dockerfile:23]: Good catch, thanks Luca

File: Makefile (hunk 1 above)
Comment (ciarams87) [reply Makefile:125]: Yes, yq is to yaml what jq is to json, exactly. I call out it's requirements in the description of this PR: `The openshift-deploy Make target requires yq to be installed (see https://github.com/mikefarah/yq). ` - I'll update the docs when the time comes but we have a separate task for that - also, it is expected that most people will still install the operator through the catalog and this Make target is probably going to be for development only, so I don't think it's unreasonable to add another dependency.

Comment (lucacome) [reply Makefile:125]: I think I had the page open before you updated the description, so I didn't see it. Sounds good.

**Review by lucacome — APPROVED** [review]

**Pull request merged**